drivers/display/display_rm68200.c

```diff
@@ -0,0 +1,1367 @@
+		if (ret < 0) {
+			LOG_ERR("Could not configure reset GPIO (%d)", ret);
+			return ret;
+		}
+		ret = gpio_pin_configure_dt(&config->pwr_gpio, GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Could not configure power GPIO (%d)", ret);
+			return ret;
+		}
+
+		/* pwr:1 -> reset:0 -> reset:1 */
+		gpio_pin_set_dt(&config->pwr_gpio, 1);
+		k_sleep(K_MSEC(1));
+		gpio_pin_set_dt(&config->reset_gpio, 0);
+		k_sleep(K_MSEC(1));
+		gpio_pin_set_dt(&config->reset_gpio, 1);
+		k_sleep(K_MSEC(5));
+	}
+
+	mdev.data_lanes = config->num_of_lanes;
+	mdev.pixfmt = config->pixel_format;
+
+	ret = mipi_dsi_attach(config->mipi_dsi, config->channel, &mdev);
+	if (ret < 0) {
+		LOG_ERR("Could not attach to MIPI-DSI host");
+		return ret;
+	}
+
+	param[0] = DCS_CMD_PAGE;
+	param[1] = DCS_CMD_PAGE_SET_PAGE0;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_EXT_PWR_IC_TYPE;
+	param[1] = 0xC0;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_EXT_PWR_SET_AVDD;
+	param[1] = 0x53;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_AVEE_FROM_PFM;
+	param[1] = 0x00;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_BT2CTR;
+	param[1] = 0xE5;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_AVDD_FROM_PFM;
+	param[1] = 0x0A;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_SETAVEE;
+	param[1] = 0x0A;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_SGOPCTR;
+	param[1] = 0x52;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_BT3CTR;
+	param[1] = 0x53;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_BT4CTR;
+	param[1] = 0x5A;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_INVCTR;
+	param[1] = 0x00;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_STBCTR;
+	param[1] = 0x0A;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_SDCTR;
+	param[1] = 0x06;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_VCMCTR;
+	param[1] = 0x56;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_SETVGMN;
+	param[1] = 0xA0;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_SETVGSN;
+	param[1] = 0x00;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_SETVGMP;
+	param[1] = 0xA0;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_SETVGSP;
+	param[1] = 0x00;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_SW_CTRL;
+	param[1] = 0x10U | (config->num_of_lanes - 1U);
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = DCS_CMD_PAGE;
+	param[1] = DCS_CMD_PAGE_SET_PAGE2;
+	rm68200_generic_write(dev, param, 2);
+
+	/* There is no description for the below registers in the datasheet */
+	param[0] = 0x00;
+	param[1] = 0x05;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = 0x02;
+	param[1] = 0x0B;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = 0x03;
+	param[1] = 0x0F;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = 0x04;
+	param[1] = 0x7D;
+	rm68200_generic_write(dev, param, 2);
```

Comment:
it's a shame that vendor requires all these generic writes and doesn't document what they mean, maybe it's a kind of binary blob, who knows! Why not create a static const array holding all "magic numbers" and dump them in a loop, at least ugliness will be concentrated in a single array...

Reply:
I have switched to using DCS writes. We had initially implemented using a static const array, later switched to this way per feedback received. This also matches the Linux driver implementation.